PluginExamples/Package.swift

```diff
@@ -1,4 +1,4 @@
-// swift-tools-version: 5.6
+// swift-tools-version: 5.8
```

Comment:
@Skoti can you please remove the `name` argument from the `swift-protobuf` package declaration when you list it as a dependency below? Like so:

```swift
let package = Package(
    name: "PluginExamples",
    dependencies: [
        .package(path: "../")
    ],
```

Looks like SPM in 5.8 is having trouble finding products when specifying a package name.
If you remove the `name` argument, it will default the package name to match the directory's name, which in this case is also `swift-protobuf` so it should all just work.
This should fix the broken CI.

Reply:
@Skoti just a friendly ping here :) you think you can take a shot at this soon? You're very close to landing the PR and it would be nice to get this change done so it can be merged.

Reply:
@gjcairo done

Reply:
It didn't help :(
now all 3 fail even though it works for me locally for Swift 5.10 🤷‍♂️

Reply:
Ah, didn't realise we are setting the path to be `/swift-protobuf/main` in the checkout GH action: https://github.com/apple/swift-protobuf/blob/main/.github/workflows/build.yml#L43-L46

Can we get rid of the `path:` there? It's unnecessary. The build should work without this.

Reply:
Yes sorry I didn't mean to do that in this PR, I should have probably tagged you @thomasvl - it was a general question.
I think we should be okay with removing the path as it's not required and I can't see it conflicting with anything else.

I can open a PR with this change and @Skoti can rebase once that's landed and get this merged.

Reply:
@thomasvl  https://github.com/apple/swift-protobuf/pull/1689

Reply:
The access-level examples need to be put into a separate job that won't run for Swift 5.8.

The current setup is that `ExampleTests` module depends on each test-case module, so it will try to compile `AccessLevelOnImport` module, and explicit access levels won't compile in Swift 5.8.

This seems like more work that should be part of a separate PR. Is it ok if we strip out plugin examples from this PR?

Reply:
> This seems like more work that should be part of a separate PR. Is it ok if we strip out plugin examples from this PR?

Fine by me, this fear was what I tried to alluded to before when I wasn't sure if the examples could be used because of it having to work with older versions.

(a `#if swift` within the Package.swift for the examples _might_ work, but that's potentially getting into the _harder to maintain_ space)

Reply:
Yeah I was just writing that you could add the new `AccessLevelOnImport` module as a target conditionally based on the compiler version (`#if compiler(>=5.9)`), **or** wrap all the source files in that module with that same compiler flag so they would be effectively empty in 5.8.
I don't think it's harder to maintain necessarily given we'll stop supporting 5.8 in a short few months when 6.0 is out anyways.
You can split this up in a separate PR if you prefer but I think doing that isn't much additional work if you just want to get it all done here.

Reply:
> Yeah I was just writing that you could add the new AccessLevelOnImport module as a target conditionally based on the compiler version (#if compiler(>=5.9)), or wrap all the source files in that module with that same compiler flag so they would be effectively empty in 5.8.

Now it should all work. Thanks!